test/integration/cni/pod_egress_traffic_test.go

```diff
@@ -0,0 +1,286 @@
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration/common"
+)
+
+const (
+	cmdForPodEgressIpv6 = `ip -f inet6 address show dev v6if0 | sed -n "s/.*inet6 \(fd00::ac:.*\)\/118.*/\1/p"`
+	cmdForPodEgressIpv4 = `ip -f inet address show dev v4if0 | sed -n "s/.*inet \(169.254..*\)\/22.*/\1/p"`
+)
+
+// Verifies connectivity to deployment behind different service types
+var _ = Describe("[CANARY] test cluster egress connectivity", func() {
+	var (
+		err error
+		// test container that verifies connectivity to en external IPv6 using IPv6 only
+		testerContainer coreV1.Container
+
+		// Primary node busybox deployment
+		primaryNodeDeployment *v1.Deployment
+		// Secondary node busybox deployment
+		secondaryNodeDeployment *v1.Deployment
+
+		primaryNodeIP   *string
+		secondaryNodeIP *string
+
+		isIPv4Cluster bool
+
+		// pods in running primary node
+		podsInPrimaryNode coreV1.PodList
+		// pods in running secondary node
+		podsInSecondaryNode coreV1.PodList
+		// IPs of PODs running in primary node
+		podsInPrimaryNodeIPs []string
+		// IPs of PODs running in secondary node
+		podsInSecondaryNodeIPs []string
+	)
+
+	BeforeEach(func() {
+		// initialize vars
+		err = nil
+		primaryNodeIP = nil
+		secondaryNodeIP = nil
+		testerContainer = coreV1.Container{}
+
+		// determine v4 or v6 cluster
+		clusterOutput, err := f.CloudServices.EKS().DescribeCluster(f.Options.ClusterName)
+		Expect(err).NotTo(HaveOccurred())
+		if *clusterOutput.Cluster.KubernetesNetworkConfig.IpFamily != "ipv6" {
+			isIPv4Cluster = true
+		}
+
+		// determine primary and secondary nodes IPs
+		PrimaryNodeInstance, err := f.CloudServices.EC2().DescribeInstance(k8sUtils.GetInstanceIDFromNode(primaryNode))
+		Expect(err).ToNot(HaveOccurred())
+		SecondaryNodeInstance, err := f.CloudServices.EC2().DescribeInstance(k8sUtils.GetInstanceIDFromNode(secondaryNode))
+		Expect(err).ToNot(HaveOccurred())
+		if isIPv4Cluster {
+			// Ipv6 address in node primary ENI is used for Ipv6 egress
+			primaryNodeIP = PrimaryNodeInstance.Ipv6Address
+			secondaryNodeIP = SecondaryNodeInstance.Ipv6Address
+		} else {
+			primaryNodeIP = PrimaryNodeInstance.PublicIpAddress
+			secondaryNodeIP = SecondaryNodeInstance.PublicIpAddress
+		}
+		Expect(primaryNodeIP).NotTo(BeNil())
+		Expect(secondaryNodeIP).NotTo(BeNil())
+
+		// initialize busybox container for testing later
+		testerContainer = manifest.NewCurlContainer().
+			Command([]string{"sleep", "3600"}).Build()
+
+		testerContainer.SecurityContext = &coreV1.SecurityContext{
+			RunAsUser:  aws.Int64(0),
+			Privileged: aws.Bool(true)} // ping needs privileged to run
+
+		// in IPv6 cluster, only need 2 pods for testing
+		var replicas = 2
+		if isIPv4Cluster {
+			// v6 cluster needs more pods to trigger secondary ENI
```

Comment:
Egress is expected to work no matter which ENI pods are behind. The case here is that even PODs are behind secondary ENI, it will still use primary ENI for egress traffic which is part of the requirements.